### DISCUSSION TITLE
Suggest a common metadata format for images the various UI developers could follow

### DISCUSSION THREAD

**Discussion by worthlesscog** [civitai/civitai#606] [Ideas]
Civit should suggest a common metadata format for image tagging including what format it should be and which tags to put it in. Currently there seem to be various different formats and placement of tags across image formats.

Civit has probably seen and parsed an awful lot of image metadata so is ideally placed to come up with some kind of schema, JSON or otherwise, that covers what's out there.

If there was a "standard" metadata format the different UIs could easily follow it meaning their images would be read properly on the site. Presumably the folks posting their work would want that. It would also stop you guys chasing changes in meta format and the need to parse the various different schemes that seem to be kicking about.

While generally the metadata seems to be extracted successfully, there are still instances where the fields end up mangled with bad quotes or bracketing.

**Comment by JustMaier** [comment]
This would be great. I've actually started to talk with a few different people about this exact thing as more people have started to use ComfyUI which doesn't provide metadata in the same way as Automatic1111. While I could throw out a few ideas and try to define this on my own, I think I'd prefer to gather a few of the tool creators in the space to talk about this.

I'll follow-up with you when I have a chance to start that.

**Comment by hipsterusername** [comment]
@JustMaier - We've been working on developing a relatively basic/flexible core metadata model ourselves. Proposing a slice of this here, and will share elsewhere as well.

![image](https://github.com/civitai/civitai/assets/31807370/8674adf9-ec45-4049-8f52-fe8c0fd63a66)

**Reply by worthlesscog** [reply]
@hipsterusername - please consider adding model hashes to your metadata model.

Model names are a mish mash with no real scheme and are really only useful in a fuzzy match to give you a model "family" (eg. Deliberate, GhostMix, DreamShaper ...) whereas a hash can give you the exact file and, by extension, the model name.

**Reply by hipsterusername** [reply]
We have always calculated the sha256, will discuss with team about incorporating it. Thanks for the suggestion!
